typescript/testsuite/api/api_service_client/api_client.ts

```diff
@@ -0,0 +1,181 @@
+//TODO - extract this to example-microservices repo
+
+import { Result, ok, err } from "neverthrow";
+import * as httpStatusCode from "http-status-codes";
+import * as axios from "axios";
+
+const PERSON_ENDPOINT: string = "person";
+
+const INCREMENT_BOOKS_READ_ENDPOINT: string = "incrementBooksRead";
+
+const HEALTHCHECK_URL_SLUG: string = "health";
+const HEALTHY_VALUE: string = "healthy";
+
+export interface Person {
+    readonly booksRead: number;
+}
+
+export class APIClient {
+    private readonly ipAddr: string;
+    private readonly port: number;
+
+    constructor (ipAddr: string, port: number) {
+        this.ipAddr = ipAddr;
+        this.port = port;
+    }
+    public async addPerson(id: number): Promise <Result<null, Error>> {
+        const url: string = this.getPersonUrlForId(id);
+        let resp: axios.AxiosResponse<any>;
+        try {
+            resp = await axios.default.post(url, null);
+        } catch(exception: any) {
+            // Sadly, we have to do this because there's no great way to enforce the caught thing being an error
+            // See: https://stackoverflow.com/questions/30469261/checking-for-typeof-error-in-js
+            if (exception && exception.stack && exception.message) {
+                return err(exception as Error);
+            }
+            return err(new Error("Performing a post request threw an exception, but " +
+                "it's not an Error so we can't report any more information than this"));
+        }
+
+        return ok(null);
+    }
+
+    public async getPerson(id: number): Promise<Result<Person, Error>> {
+        const url: string = this.getPersonUrlForId(id);
+        let resp: axios.AxiosResponse<any>;
+        try {
+            resp = await axios.default.get(url);
+        } catch(exception: any) {
+            // Sadly, we have to do this because there's no great way to enforce the caught thing being an error
+            // See: https://stackoverflow.com/questions/30469261/checking-for-typeof-error-in-js
+            if (exception && exception.stack && exception.message) {
+                return err(exception as Error);
+            }
+            return err(new Error("Performing a get request threw an exception, but " +
+                "it's not an Error so we can't report any more information than this"));
+        }
+
+        const body: any = resp.data;
+        
+        let bodyString: string;
+        try {
+            bodyString = JSON.stringify(body);
```

Comment:
Why are we `JSON.stringify`'ing the response data?? Shouldn't we just be parsing it?

Reply:
"JSON.stringify() method converts a JavaScript object or value to a JSON string" which we then need for JSON.parse() below which "parses a JSON string, constructing the JavaScript value or object described by the string"